pkg/cmd/dashboard.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"context"
+	"flag"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/cloud-native-application/rudrx/api/types"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	cmdutil "github.com/cloud-native-application/rudrx/pkg/cmd/util"
+	"github.com/cloud-native-application/rudrx/pkg/server"
+	"github.com/spf13/cobra"
+)
+
+func NewDashboardCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
+	// ctx := context.Background()
+	cmd := &cobra.Command{
+		Use:     "dashboard",
+		Short:   "Setup API Server and launch Dashboard",
+		Long:    "Setup API Server and launch Dashboard",
+		Example: `dashboard`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			newClient, err := client.New(c.Config, client.Options{Scheme: c.Schema})
+			if err != nil {
+				return err
+			}
+			SetupApiServer(newClient)
+			return nil
+		},
+		Annotations: map[string]string{
+			types.TagCommandType: types.TypeSystem,
+		},
+	}
+	cmd.SetOut(ioStreams.Out)
+	return cmd
+}
+
+/*
+SetupApiServer moved main function from cmd/server/main.go written by Ryan, so it can be integrated
+by Cli and called by cmd/server/main.go.
+*/
+func SetupApiServer(kubeClient client.Client) {
+	var logFilePath string
+	var logRetainDate int
+	var logCompress, development bool
+
+	flag.StringVar(&logFilePath, "log-file-path", "", "The log file path.")
+	flag.IntVar(&logRetainDate, "log-retain-date", 7, "The number of days of logs history to retain.")
+	flag.BoolVar(&logCompress, "log-compress", true, "Enable compression on the rotated logs.")
```

Comment:
we don't need these arguments for command line, just use stdout

Reply:
this is for the RESTful server logging, won't this be good to have? We don't want to print those logs to the stdout, right?

Reply:
If this server is deployed in K8s cluster, print to stdout is enough.
If this is used as cli dashboard command, print to sdtout is also fine.

Reply:
I don't get it, users don't want to see our server log printing out to their Cli command line, right?